dipy/align/imaffine.py

```diff
@@ -222,8 +225,9 @@ def _apply_transform(self, image, interp='linear', image_grid2world=None,
         shape = np.array(sampling_grid_shape, dtype=np.int32)
 
         # Verify valid image dimension
-        if dim < 2 or dim > 3:
-            raise ValueError('Undefined transform for dimension: %d' % (dim,))
+        img_dim = len(image.shape)
```

Comment:
Sorry to be lazy - so this was previously a bug right?  In that you were checking the output dims rather than the input dims?  Do the current tests check for this?